File: Backend/src/Meseta_Verda.Domain/Entities/Categoria.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Meseta_Verda.Domain.Entities
{
    public class Categoria
    {
        [Key]
        public int IdCategoria { get; set; }
        
        [Required]
        [MaxLength(100)]
        public string Nombre { get; set; } = null!;
        
        public ICollection<Producto> Productos { get; set; } = new List<Producto>();
    }
}

File: Frontend/agrotrade_frontend/lib/models/Productor/catalogo/categoria.dart
class Categoria {
  final int idCategoria;
  final String nombre;

  const Categoria({
    required this.idCategoria,
    required this.nombre,
  });

  factory Categoria.fromJson(Map<String, dynamic> json) {
    return Categoria(
      idCategoria: json['idCategoria'] as int,
      nombre: json['nombre'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idCategoria': idCategoria,
      'nombre': nombre,
    };
  }
}
